compass/milestone/MilestoneDatabase.java

```diff
@@ -75,18 +76,46 @@ public MilestoneDatabase(SpongeFactory.Mode powMode, SignatureSource signatureSo
     this.powMode = powMode;
   }
 
+  private static List<String> readInBatches(BufferedReader br, int batchSize, int totalSize) throws IOException {
+    List<String> result = new ArrayList<>(totalSize);
+    for (int i = 1; i < totalSize + batchSize; i += batchSize) {
+      List<String> batch = readLinesInBatches(br, batchSize);
+      result.addAll(batch);
+      if (batch.size() < batchSize) {
+        return result;
+      }
+    }
+    return result;
+  }
+
+  private static List<String> readLinesInBatches(BufferedReader br, int batchSize) throws IOException {
+    List<String> result = new ArrayList<>(batchSize);
+    for (int i = 1; i < batchSize; i++) {
```

Comment:
Is the break condition of the `for` loop correct? Take this edge case: if `batchSize` is 1 no line gets read. Also, if `i` represents the amount of line read, why not starting it at `0`?